common/display/displayqueue.cpp

```diff
@@ -696,6 +696,11 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
     }
   }
 
+  if (has_video_layer && video_effect_changed_) {
```

Comment:
This should be split into two separate "if" statements since it is highly unlikely that both flags will be set concurrently.